feat: add ansi highlighting support

blake-mealey opened the pull request:
I recently added ANSI highlighting to shiki (see https://github.com/shikijs/shiki/pull/386/) and I'd like to make it available through rehype-pretty-code as well. Since shiki is a peer dep, I check for the existence of the `ansiToHtml` function before using.

atomiks commented on website/pages/index.mdx:
````diff
@@ -79,7 +79,6 @@ function MyComponent() {
   );
 }
 ```
-
````
Intentional?

atomiks commented on test/fixtures/ansi.md:
````diff
@@ -0,0 +1,12 @@
+## ANSI
+
+```ansi filename="vite --port 3123"
````
This lib uses `title="..."`

atomiks reviewed (commented):
Seems like the tests are all failing

blake-mealey commented:
I'm getting test failures on master, not sure what's going on :/

blake-mealey commented:
I've fixed the other comments

blake-mealey commented:
OK after looking into this a bit more, it's the upgraded shiki version which is causing the issues. I'll dig a little deeper.

blake-mealey commented:
Fixed! There were a couple of things going on:

1. The shiki version introduced some changes to the colors in the theme the tests are using (I've updated the snapshots)
2. I had a bug which caused block colors to be incorrect (I've fixed this in the latest commit)

atomiks commented:
Looks like only one test is failing, a new one introduced the other day

blake-mealey commented:
Hmm, when I run the tests locally, that test succeeds. I manually updated the snapshot file so that it passes in CI, but I'm not sure what the difference is. Locally it adds a `tabindex="0"` attribute to the `pre` tag for some reason. None of the other files have tabindex :/

blake-mealey commented:
Mind taking another look at this one? I believe the website builds are failing because it is not using the new version of the package (since it isn't released yet). Otherwise tests are passing

atomiks commented:
Forgot about this sorry, looks good :)

aryadaroui commented:
> Hmm, when I run the tests locally, that test succeeds. I manually updated the snapshot file so that it passes in CI, but I'm not sure what the difference is. Locally it adds a `tabindex="0"` attribute to the `pre` tag for some reason. None of the other files have tabindex :/

Do you know why the `tab-index="0"` attribute is being added? It throws accessibility a11y warnings:
```
A11y: noninteractive element cannot have nonnegative tabIndex value
```

atomiks commented:
> Do you know why the tab-index="0" attribute is being added? It throws accessibility a11y warnings:

I think it may be a bug with `shiki@0.14.1`+ as it doesn't happen with `shiki@0.14.0`.
